src/main/java/org/opensearch/security/resources/api/migrate/MigrateResourceSharingInfoApiAction.java

```diff
@@ -201,7 +225,19 @@ private ValidationResult<Triple<String, String, List<SourceDoc>>> loadCurrentSha
                     }
                 }
 
-                results.add(new SourceDoc(id, username, backendRoles));
+                String type = null;
+
+                if (typePath != null) {
+                    type = rec.at(typePath.startsWith("/") ? typePath : ("/" + typePath)).asText(null);
+                    if (type == null) {
+                        LOGGER.debug("Record without associated type, skipping entirely: {}", hit.getId());
```

Comment:
This should be added to the skippedCount.

Reply:
Done. I moved this logic down below near the username logic as well.